src/mlpack/methods/ann/layer/upsample.hpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * @file methods/ann/layer/upsample.hpp
+ * @author Abhinav Anand
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license. You should have received a copy of the
+ * 3-clause BSD license along with mlpack. If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_BILINEAR_INTERPOLATION_HPP
+#define MLPACK_METHODS_ANN_LAYER_BILINEAR_INTERPOLATION_HPP
```

Comment:
I guess it makes sense if the header guard is named after the file, at least that's what we used in other places.